examples/systemcalls/systemcalls.c

```diff
@@ -45,23 +53,25 @@ bool do_exec(int count, ...)
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
 
     va_end(args);
 
-    return true;
+    //create child process
+    child_pid = fork();
+    if (child_pid == -1) {
+	return false; //child process creation failed
+    }
+    if (child_pid == 0) {
+	execv(command[0], command);
+	exit(-1);
+    } else {
+	int rc =  waitpid(child_pid, &child_status, 0);
+	if (rc != -1 && child_status == 0) {
```

Comment:
@maheshchnt you should use WIFEXETED and WEXITSTATUS on child_status here - see https://linux.die.net/man/2/waitpid